aspnetcore/security/anti-request-forgery.md

```diff
@@ -1,18 +1,311 @@
+    if (user != null)
+    {
+        var result = await _userManager.RemoveLoginAsync(user, account.LoginProvider, account.ProviderKey);
+        if (result.Succeeded)
+        {
+            await _signInManager.SignInAsync(user, isPersistent: false);
+            message = ManageMessageId.RemoveLoginSuccess;
+        }
+    }
+    return RedirectToAction(nameof(ManageLogins), new { Message = message });
+}
+```
+
+The ``ValidateAntiForgeryToken`` attribute will require a token for every request, including *GET* requests. If you apply it broadly, you can override it with the ``IgnoreAntiforgeryToken`` attribute.
+
+### AutoValidateAntiforgeryToken
+
+In most cases, your application will not receive antiforgery tokens for certain kinds of HTTP requests, such as GET requests. Instead of broadly applying the ``ValidateAntiForgeryToken`` attribute and then overriding it with ``IgnoreAntiforgeryToken`` attributes, you can use the ``AutoValidateAntiforgeryToken`` attribute. This attribute works identically to the ``ValidateAntiForgeryToken`` attribute, except that it ignores the following HTTP methods:
+
+* GET
+* HEAD
+* OPTIONS
+* TRACE
+
+The ASP.NET Core team recommends you use ``AutoValidateAntiforgeryToken`` broadly for non-API scenarios, so that your POST actions are protected by default (APIs don't have an automatic mechanism for sending the non-cookie part of the token). The alternative is to ignore antiforgery tokens by default, unless ``ValidateAntiForgeryToken`` is applied to the individual action method. It's more likely in this scenario for a POST action method to be left unprotected, leaving your app vulnerable to CSRF attacks. Even anonymous POSTS should send the antiforgery token.
+
+Example (class level):
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+[Authorize]
+[AutoValidateAntiforgeryToken]
+public class ManageController : Controller
+{
+```
+
+Example (globally):
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+services.AddMvc(options => 
+    options.Filters.Add(new AutoValidateAntiforgeryTokenAttribute()));
+```
+
+### IgnoreAntiforgeryToken
+
+The ``IgnoreAntiforgeryToken`` filter is used to eliminate the need for an antiforgery token to be present for a given action (or controller). When applied, this filter will override ``ValidateAntiForgeryToken`` and/or ``AutoValidateAntiforgeryToken`` filters specified at a higher level (globally or on a controller).
+
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+[Authorize]
+[AutoValidateAntiforgeryToken]
+public class ManageController : Controller
+{
+  [HttpPost]
```

Comment:
This particular example requires `[HttpPost]` because otherwise the `AutoValidateAntiforgeryToken` would already ignore token validation on this method (making it unnecessary to add the `IgnoreAntiforgeryToken` attribute).